src/main/java/org/sasanlabs/service/vulnerability/xss/reflected/XSSInImgTagAttributeInjectionVulnerability.java

```diff
@@ -169,17 +195,47 @@ public UrlParamBasedImgTagAttrInjection() {
             htmlTemplate = "LEVEL_1/XSS")
     public ResponseEntity<String> getVulnerablePayloadLevelSecure(
             @RequestParam Map<String, String> queryParams) {
+
         String vulnerablePayloadWithPlaceHolder = "<img src=\"%s\" width=\"400\" height=\"300\"/>";
         StringBuilder payload = new StringBuilder();
 
         for (Map.Entry<String, String> map : queryParams.entrySet()) {
-            if (allowedValues.contains(map.getValue())) {
+
+            if (isSecure(map.getValue())) {
                 payload.append(
                         String.format(
                                 vulnerablePayloadWithPlaceHolder,
                                 StringEscapeUtils.escapeHtml4(map.getValue())));
             }
         }
+
         return new ResponseEntity<String>(payload.toString(), HttpStatus.OK);
     }
+
+    private boolean isSecure(String possibleImageSourcePath) {
+
+        if (possibleImageSourcePath.startsWith("/")) {
+            return false;
+        }
+
+        if (possibleImageSourcePath.contains("..")) {
+            return false;
+        }
+
+        for (char c : possibleImageSourcePath.toCharArray()) {
+            if (isIllegalCharacter(c)) {
+                return false;
+            }
```

Comment:
This is not needed, have a look at: https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html for more details on XSS prevention.

Reply:
I think you are trying to protect against path traversal but this is not vulnerable to Path Traversal.